src/backend/gpopt/gpdbwrappers.cpp

```diff
@@ -28,28 +28,22 @@
 #include "gpos/error/CAutoExceptionStack.h"
 #include "gpos/error/CException.h"
 
-#include "gpopt/utils/gpdbdefs.h"
 #include "naucrates/exception.h"
 
-#include "catalog/pg_collation.h"
+#pragma GCC diagnostic push
```

Comment:
Seems like we're targeting on clang support as well, but only gcc in considered here. Also, I'm not sure if gcc and clang behave the same with -Wpedantic flag. Could you add clang support or justify its absence

Reply:
I've tested with clang-15 as well. Build seems to be fine.

Reply:
Seems like clang targets to fully support the code written for gcc, so you are correct, clang works with #pragma gcc too